packages/amplify-provider-awscloudformation/lib/transform-graphql-schema.js

```diff
@@ -161,12 +162,22 @@ async function transformerVersionCheck(context, resourceDir, cloudBackendDirecto
   // if we already asked the confirmation question before at a previous push
   // or during current operations we should not ask again.
   const showPrompt = !(cloudVersionExist || localVersionExist);
+  if (!showPrompt) {
+    transformerVersion = 3;
+  } else if (localTransformerConfig.Version) {
+    transformerVersion = localTransformerConfig.Version;
+  }
+  // get message for the transformer if it exists
+  if (transformerVersion && TRANSFORM_CONFIG_VERSION_MAP[transformerVersion]) {
+    versionChangeMessage = TRANSFORM_CONFIG_VERSION_MAP[transformerVersion](usedDirectives);
+  }
 
   const resources = updatedResources.filter(resource => resource.service === 'AppSync');
 
-  if (showPrompt && usedDirectives.includes('auth') && resources.length > 0) {
+  if (versionChangeMessage && resources.length > 0) {
     if (context.exeInfo && context.exeInfo.inputParams && context.exeInfo.inputParams.yes) {
       context.print.warning(`\n${versionChangeMessage}\n`);
+      transformerVersion += 1;
```

Comment:
This logic is not scalable. If we have another bump in the transform conf version then this logic would need a big refactor. I would recommend putting all the warning messages in an array based on logic on transform conf version - after which you just have one question to ask in the flow - "Do you want to continue?"

Reply:
Feature flags are the way to solve this long term

Reply:
We should not just increment this. We should know exactly what version of transformer we are at rather then incrementing it